ring/func/base.py

```diff
@@ -13,6 +13,11 @@
 from ..key import CallableKey
 from ..coder import registry as default_registry
 
+try:
+    import numpy as np
```

Comment:
Though `np` is a common abbr for numpy applications, using `numpy` will be better here because this is actually not a numpy app.